consul/resource_consul_service.go

```diff
@@ -594,6 +601,11 @@ func getCatalogRegistration(d *schema.ResourceData, meta interface{}) (*consulap
 	}
 	registration.Service.Meta = serviceMeta
 
+	if serviceWeights, ok := d.GetOk("weights"); ok {
+		registration.Service.Weights.Passing = serviceWeights.(map[string]interface{})["passing"].(int)
+		registration.Service.Weights.Warning = serviceWeights.(map[string]interface{})["warning"].(int)
```

Comment:
Need to check if these are set, just because we have `weights` doesn't mean these exist.